packages/react/src/components/dropzone.tsx

```diff
@@ -106,15 +106,15 @@ export function UploadDropzone<TRouter extends FileRouter>(
 
   const onDrop = useCallback(
     (acceptedFiles: FileWithPath[]) => {
+      if ($props.onBeforeUploadBegin) {
```

Comment:
do you want this to run on every drop? The naming doesn't really tell that it will

Reply:
from what I can tell it needs to run before `files` gets set for the rename to be "complete" on all the callbacks like `onUploadComplete`, so with that in mind I added it here.

in [other libs](https://github.com/odysseyscience/react-s3-uploader/blob/master/s3upload.js#L34-L36) I've seen similar functions called `scrubFilename` which might be more accurate in terms of intent versus describing when it will upload